eris.go

```diff
@@ -72,10 +74,35 @@ func wrap(err error, msg string) error {
 			root.stack.insertPC(*stack)
 		}
 	default:
+		// attempt to unwrap external errors while building a new error chain or fallback to flattening them
+		var errStr []string
+		for e != nil {
+			str := e.Error()
+			errStr = append([]string{str}, errStr...)
+			e = Unwrap(e)
+			// unwrap twice for pkg/errors and other libraries like it
+			if e != nil && e.Error() == str {
+				e = Unwrap(e)
+			}
+		}
 		err = &rootError{
-			msg:   e.Error(),
+			msg:   errStr[0],
 			stack: stack,
 		}
+		for i := 1; i < len(errStr); i++ {
```

Comment:
we can potentially put this parsing in condition if type of err is pkg/error type

Reply:
the parsing isn't just for pkg/errors types, it's for all external errors that can be unwrapped